src/transformers/models/splinter/modeling_splinter.py

```diff
@@ -0,0 +1,1556 @@
+
+        self.dense = nn.Linear(self.input_dim, self.output_dim)
+        self.act_fn = ACT2FN[hidden_act]
+        self.LayerNorm = nn.LayerNorm(self.output_dim)
+
+    def forward(self, inputs):
+        temp = self.dense(inputs)
+        temp = self.act_fn(temp)
+        temp = self.LayerNorm(temp)
+        return temp
+
+
+class QuestionAwareSpanSelectionHead(nn.Module):
+    """
+    Implementation of Question-Aware Span Selection (QASS) head, described in Splinter's paper:
+
+    """
+
+    def __init__(self, config):
+        super().__init__()
+
+        self.query_start_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.query_end_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.start_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+        self.end_transform = FullyConnectedLayer(config.hidden_size, config.hidden_size)
+
+        self.start_classifier = nn.Linear(config.hidden_size, config.hidden_size, bias=False)
+        self.end_classifier = nn.Linear(config.hidden_size, config.hidden_size, bias=False)
+
+    def forward(self, inputs, positions):
+        gathered_reps = gather_positions(inputs, positions)
+
+        query_start_reps = self.query_start_transform(gathered_reps)  # [batch_size, num_positions, dim]
+        query_end_reps = self.query_end_transform(gathered_reps)  # [batch_size, num_positions, dim]
+        start_reps = self.start_transform(inputs)  # [batch_size, seq_length, dim]
+        end_reps = self.end_transform(inputs)  # [batch_size, seq_length, dim]
+
+        temp = self.start_classifier(query_start_reps)  # [batch_size, num_positions, dim]
+        start_reps = start_reps.permute(0, 2, 1)  # [batch_size, dim, seq_length]
+        start_logits = torch.matmul(temp, start_reps)
+
+        temp = self.end_classifier(query_end_reps)
+        end_reps = end_reps.permute(0, 2, 1)
+        end_logits = torch.matmul(temp, end_reps)
+
+        return start_logits, end_logits
+
+
+class ClassificationHead(nn.Module):
+    # TODO(Ori/Yuval): document
+    def __init__(self, config):
+        super().__init__()
+        self.span_predictions = QuestionAwareSpanSelectionHead(config)
+
+    def forward(self, inputs, positions):
+        return self.span_predictions(inputs, positions)
+
+
+@add_start_docstrings(
+    """
+    Splinter Model with a span classification head on top for extractive question-answering tasks like SQuAD (a linear
+    layers on top of the hidden-states output to compute `span start logits` and `span end logits`).
+    """,
+    SPLINTER_START_DOCSTRING,
+)
+class SplinterForQuestionAnswering(SplinterPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+
+        self.splinter = SplinterModel(config)
+        self.initialize_new_qass = config.initialize_new_qass
+        self.splinter_qass = QuestionAwareSpanSelectionHead(config) if not self.initialize_new_qass else None
```

Comment:
I don't really understand why we have `self.splinter_qass` which is the exact same module as `self.new_splinter_qass` and it's always only one of the 2 -> why not just have one self.splinter_qass`?

Reply:
Is it because of the naming in the weights?

Reply:
Exactly. This was our way to differentiate between keeping the weights of this layer and initializing them randomly. 
Any suggestions will be welcome, it's pretty ugly :)

Reply:
By the way, is there any way we can do it while keeping the logging (which of the weights from the checkpoint are not used and vice versa) correct?

Reply:
Yeah in this case, we should have multiple checkpoints I think.
One checkpoint which **does have** the trained weights for `spliter_qass` and one checkpoint which does not have them. For the checkpoint that does not have the weights for `spliter_qass` they will automatically be randomely initialized

Reply:
Great.. So I think the way to control it will be via `--model_name_or_path`, right?
In other words, I'll remove it from `SplinterConfig`.
Unless you had other thoughts?